articles/140_topic_and_service_name_mapping.md

```diff
@@ -211,23 +194,51 @@ For example, given the name `{private}foo`, if there is a substitution called `{
+The DDS partition will contain the ROS name's namespace, including any forward slashes (`/`) that make up the namespace, and that is acceptable because DDS partitions allow forward slashes (`/`) unlike the DDS topics.
+
+This strategy also avoids DDS partitions from being exposed to the ROS API's.
+This is because the splitting of the ROS name into namespace and base name can occur inside the implementation of the ROS middleware interface, and so the ROS topic or service name will remain intact throughout the middleware agnostic code.
+This means that if someone chooses to implement ROS 2 on top of something other than DDS, then this strategy will not impact that implementation, as it applies only to DDS implementations.
 
-### Substitution of the Namespace Delimiter
+#### DDS Partitions Characteristics
 
-The namespace delimiter in ROS 2 topic and service names, a forward slash (`/`), will be replaced with double underscores (`__`).
-Note that as fully qualified ROS 2 topic and service names are absolute, there is always a leading forward slash (`/`).
+DDS partitions are implemented as an array of strings within the `DDS::Publisher` and `DDS::Subscriber` QoS settings.
+The array allows up to 64 items, and each string in the array is an individual partition.
+This means that every string of the array is an independently matchable entity when paired with the topic name and the other QoS settings.
+Put another way, a DDS DataWriter and DataReader will not connect to one another unless at least one of the partition strings in the array match, even if the topic names match.
+Sometimes special characters are used in the partition strings to express regular expressions matching (such as `+`, `*`, `^`), but they are not officially supported by the standard.
+The maximum size of each string in the array may depend on the DDS or RTPS implementation.
+You can read more about partitions in RTI's documentation:
 
-### ROS Specific Name Prefix
+- [PARTITION_QosPolicy](https://community.rti.com/static/documentation/connext-dds/5.2.3/doc/manuals/connext_dds/html_files/RTI_ConnextDDS_CoreLibraries_UsersManual/Content/UsersManual/PARTITION_QosPolicy.htm)
 
-In order to differentiate ROS topics easily, all DDS topic names created by ROS shall be prefixed with `rX`, where `X` is a single character that indicates to which subsystem of ROS the topic belongs.
-For example, a plain topic called `/foo` would translate to a DDS topic named `rt__foo`, which is the result of concatenating the prefix `rt` for being a ROS topic, with `__` for the leading `/`, and the topic name `foo`.
+The array of strings have no hierarchy and the order does not matter.
+Each entry in the partition array is directly combined with the DDS topic and they are not sequentially combined.
+If a publisher operates on two partition entries, e.g. `foo` and `bar` with a base name of `baz`, this would result in the moral equivalent of two topics: `/foo/baz` and `/bar/baz`
```

Comment:
A phrase I've shamelessly adopted from @gerkey. It just means "a la" or "in the spirit of". I'll just reword it to avoid any turn of phrase. 401d5bb